<?xml version="1.0" encoding="utf-8"?>
<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.2" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
  <file datatype="xml" original="NullReferenceException.xml" source-language="en-US" target-language="pt-BR">
    <header>
      <tool tool-id="mdxliff" tool-name="mdxliff" tool-version="1.0-15c36f0" tool-company="Microsoft" />
      <xliffext:skl_file_name xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">02cd5861-7ce2-4a82-b358-31f8435a0ac5871adebd50877699047b62f9d350cf2555dfe480.skl</xliffext:skl_file_name>
      <xliffext:version xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">1.2</xliffext:version>
      <xliffext:ms.openlocfilehash xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">871adebd50877699047b62f9d350cf2555dfe480</xliffext:ms.openlocfilehash>
      <xliffext:ms.sourcegitcommit xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b</xliffext:ms.sourcegitcommit>
      <xliffext:ms.lasthandoff xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">04/03/2018</xliffext:ms.lasthandoff>
      <xliffext:moniker_ids xmlns:xliffext="urn:microsoft:content:schema:xliffextensions">netcore-1.0,netcore-1.1,netcore-2.0,netcore-2.1,netframework-4.5.1,netframework-4.5.2,netframework-4.5,netframework-4.6.1,netframework-4.6.2,netframework-4.6,netframework-4.7.1,netframework-4.7,netstandard-1.0,netstandard-1.1,netstandard-1.2,netstandard-1.3,netstandard-1.4,netstandard-1.5,netstandard-1.6,netstandard-2.0,xamarinandroid-7.1,xamarinios-10.8,xamarinmac-3.0</xliffext:moniker_ids>
    </header>
    <body>
      <group id="content" extype="content">
        <trans-unit id="101" translate="yes" xml:space="preserve" uid="T:System.NullReferenceException">
          <source>The exception that is thrown when there is an attempt to dereference a null object reference.</source>
          <target state="translated">A exceção que é gerada quando ocorre uma tentativa de desreferenciar um objeto de referência nula.</target>       </trans-unit>
        <trans-unit id="102" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception is thrown when you try to access a member on a type whose value is <ph id="ph2">`null`</ph>.</source>
          <target state="translated">Um <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exceção é gerada quando você tentar acessar um membro de um tipo cujo valor é <ph id="ph2">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="103" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception typically reflects developer error and is thrown in the following scenarios:</source>
          <target state="translated">Um <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exceção normalmente reflete o erro do desenvolvedor e é gerada nas seguintes situações:</target>       </trans-unit>
        <trans-unit id="104" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You've forgotten to instantiate a reference type.</source>
          <target state="translated">Você esqueceu de instanciar um tipo de referência.</target>       </trans-unit>
        <trans-unit id="105" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>In the following example, <ph id="ph1">`names`</ph> is declared but never instantiated:</source>
          <target state="translated">No exemplo a seguir, <ph id="ph1">`names`</ph> está declarada mas nunca é instanciado:</target>       </trans-unit>
        <trans-unit id="106" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Some compilers issue a warning when they compile this code.</source>
          <target state="translated">Alguns compiladores emitirá um aviso quando eles compilar seu código.</target>       </trans-unit>
        <trans-unit id="107" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Others issue an error, and the compilation fails.</source>
          <target state="translated">Outros emitirá um erro e a compilação falha.</target>       </trans-unit>
        <trans-unit id="108" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>To address this problem, instantiate the object so that its value is no longer <ph id="ph1">`null`</ph>.</source>
          <target state="translated">Para resolver esse problema, instanciar o objeto para que seu valor não é mais <ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="109" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following example does this by calling a type's class constructor.</source>
          <target state="translated">O exemplo a seguir faz isso chamando o construtor da classe do tipo.</target>       </trans-unit>
        <trans-unit id="110" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You've forgotten to dimension an array before initializing it.</source>
          <target state="translated">Você esqueceu para uma matriz de dimensão antes de inicializar a ele.</target>       </trans-unit>
        <trans-unit id="111" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>In the following example, <ph id="ph1">`values`</ph> is declared to be an integer array, but the number of elements that it contains is never specified.</source>
          <target state="translated">No exemplo a seguir, <ph id="ph1">`values`</ph> é declarado como uma matriz de inteiros, mas o número de elementos que ele contém nunca é especificado.</target>       </trans-unit>
        <trans-unit id="112" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The attempt to initialize its values therefore thrown a  <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">A tentativa de inicializar seus valores, portanto, é geradas uma <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exceção.</target>       </trans-unit>
        <trans-unit id="113" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You can eliminate the exception by declaring the number of elements in the array before initializing it, as the following example does.</source>
          <target state="translated">Você pode eliminar a exceção declarando o número de elementos na matriz antes da inicialização, como o exemplo a seguir.</target>       </trans-unit>
        <trans-unit id="114" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For more information on declaring and initializing arrays, see <bpt id="p1">[</bpt>Arrays<ept id="p1">](http://msdn.microsoft.com/library/bb79bdde-e570-4c30-adb0-1dd5759ae041)</ept> and <bpt id="p2">[</bpt>Arrays<ept id="p2">](http://msdn.microsoft.com/library/dbf29737-b589-4443-bee6-a27588d9c67e)</ept>.</source>
          <target state="translated">Para obter mais informações sobre declaração e inicialização de matrizes, consulte <bpt id="p1">[</bpt>matrizes<ept id="p1">](http://msdn.microsoft.com/library/bb79bdde-e570-4c30-adb0-1dd5759ae041)</ept> e <bpt id="p2">[</bpt>matrizes<ept id="p2">](http://msdn.microsoft.com/library/dbf29737-b589-4443-bee6-a27588d9c67e)</ept>.</target>       </trans-unit>
        <trans-unit id="115" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You get a <bpt id="p1">**</bpt>null<ept id="p1">**</ept> return value from a method, and then call a method on the returned type.</source>
          <target state="translated">Você obtém um <bpt id="p1">**</bpt>nulo<ept id="p1">**</ept> retornar o valor de um método e, em seguida, chamar um método no tipo retornado.</target>       </trans-unit>
        <trans-unit id="116" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>This sometimes is the result of a documentation error; the documentation fails to note that a method call can return <ph id="ph1">`null`</ph>.</source>
          <target state="translated">Isso às vezes é o resultado de um erro de documentação; a documentação não conseguir Observe que uma chamada de método pode retornar <ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="117" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>In other cases, your code erroneously assumes that the method will always return a non-<bpt id="p1">**</bpt>null<ept id="p1">**</ept> value.</source>
          <target state="translated">Em outros casos, seu código erroneamente pressupõe que o método sempre retornará não<bpt id="p1">**</bpt>nulo<ept id="p1">**</ept> valor.</target>       </trans-unit>
        <trans-unit id="118" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The code in the following example assumes that the <ph id="ph1">&lt;xref:System.Array.Find%2A?displayProperty=nameWithType&gt;</ph> method always returns <ph id="ph2">`Person`</ph> object whose <ph id="ph3">`FirstName`</ph> field matches a search string.</source>
          <target state="translated">O código no exemplo a seguir supõe que o <ph id="ph1">&lt;xref:System.Array.Find%2A?displayProperty=nameWithType&gt;</ph> método sempre retorna <ph id="ph2">`Person`</ph> do objeto cuja <ph id="ph3">`FirstName`</ph> campo corresponder uma cadeia de caracteres de pesquisa.</target>       </trans-unit>
        <trans-unit id="119" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Because there is no match, the runtime throws a <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Como não há nenhuma correspondência, o tempo de execução lança um <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exceção.</target>       </trans-unit>
        <trans-unit id="120" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>To address this problem, test the method's return value to ensure that it is not <ph id="ph1">`null`</ph> before calling any of its members, as the following example does.</source>
          <target state="translated">Para resolver esse problema, teste o valor de retorno do método para garantir que não seja <ph id="ph1">`null`</ph> antes de chamar qualquer um de seus membros, como o exemplo a seguir.</target>       </trans-unit>
        <trans-unit id="121" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You're using an expression (for example, you're chaining a list of methods or properties together) to retrieve a value and, although you're checking whether the value is <ph id="ph1">`null`</ph>, the runtime still throws a <ph id="ph2">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Você está usando uma expressão (por exemplo, você está encadear uma lista de métodos ou propriedades) para recuperar um valor e, embora você está verificando se o valor é <ph id="ph1">`null`</ph>, o tempo de execução ainda gera um <ph id="ph2">&lt;xref:System.NullReferenceException&gt;</ph> exceção.</target>       </trans-unit>
        <trans-unit id="122" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>This occurs because one of the intermediate values in the expression returns <ph id="ph1">`null`</ph>.</source>
          <target state="translated">Isso ocorre porque um dos valores intermediários na expressão retorna <ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="123" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>As a result, your rest for <ph id="ph1">`null`</ph> is never evaluated.</source>
          <target state="translated">Como resultado, o restante para <ph id="ph1">`null`</ph> nunca é calculada.</target>       </trans-unit>
        <trans-unit id="124" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following example defines a <ph id="ph1">`Pages`</ph> object that  caches information about web pages, which are presented by <ph id="ph2">`Page`</ph> objects.</source>
          <target state="translated">O exemplo a seguir define uma <ph id="ph1">`Pages`</ph> objeto que armazena informações sobre páginas da web, que são apresentados pelo <ph id="ph2">`Page`</ph> objetos.</target>       </trans-unit>
        <trans-unit id="125" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The <ph id="ph1">`Example.Main`</ph> method checks whether the current web page has a non-null title and, if it does, displays the title.</source>
          <target state="translated">O <ph id="ph1">`Example.Main`</ph> método verifica se a página da web atual tiver um título não nulo e, em caso afirmativo, exibe o título.</target>       </trans-unit>
        <trans-unit id="126" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Despite this check, however, the method throws a <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Apesar dessa verificação, no entanto, o método gera uma <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exceção.</target>       </trans-unit>
        <trans-unit id="127" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The exception is thrown because <ph id="ph1">`pages.CurrentPage`</ph> returns <ph id="ph2">`null`</ph> if no page information is  stored in the cache.</source>
          <target state="translated">A exceção é gerada porque <ph id="ph1">`pages.CurrentPage`</ph> retorna <ph id="ph2">`null`</ph> se nenhuma informação de página é armazenada em cache.</target>       </trans-unit>
        <trans-unit id="128" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>This exception can be corrected by testing the value of the <ph id="ph1">`CurrentPage`</ph> property before retrieving the current <ph id="ph2">`Page`</ph> object's <ph id="ph3">`Title`</ph> property, as the following example does:</source>
          <target state="translated">Essa exceção pode ser corrigida por testar o valor da <ph id="ph1">`CurrentPage`</ph> propriedade antes de recuperar atual <ph id="ph2">`Page`</ph> do objeto <ph id="ph3">`Title`</ph> propriedade, como o exemplo a seguir faz:</target>       </trans-unit>
        <trans-unit id="129" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You're enumerating the elements of an array that contains reference types, and your attempt to process one of the elements throws a <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Você está enumerando os elementos de uma matriz que contém os tipos de referência e a tentativa de processo de um dos elementos lança um <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exceção.</target>       </trans-unit>
        <trans-unit id="130" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following example defines a string array.</source>
          <target state="translated">O exemplo a seguir define uma matriz de cadeia de caracteres.</target>       </trans-unit>
        <trans-unit id="131" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A <ph id="ph1">`for`</ph> statement enumerates the elements in the array and calls each string's <ph id="ph2">&lt;xref:System.String.Trim%2A&gt;</ph> method before displaying the string.</source>
          <target state="translated">Um <ph id="ph1">`for`</ph> instrução enumera os elementos na matriz e chama cada cadeia de caracteres <ph id="ph2">&lt;xref:System.String.Trim%2A&gt;</ph> método antes de exibir a cadeia de caracteres.</target>       </trans-unit>
        <trans-unit id="132" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>This exception occurs if you assume that each element of the array must contain a non-null value, and the value of the array element is in fact <ph id="ph1">`null`</ph>.</source>
          <target state="translated">Essa exceção ocorre se você presumir que cada elemento da matriz deve conter um valor não nulo e o valor do elemento da matriz é na verdade <ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="133" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The exception can be eliminated by testing whether the element is <ph id="ph1">`null`</ph> before performing any operation on that element, as the following example shows.</source>
          <target state="translated">A exceção pode ser eliminada testando se o elemento é <ph id="ph1">`null`</ph> antes de executar qualquer operação no elemento, como o exemplo a seguir mostra.</target>       </trans-unit>
        <trans-unit id="134" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception is thrown by a method that is passed <ph id="ph2">`null`</ph>.</source>
          <target state="translated">Um <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exceção é lançada por um método que é passado <ph id="ph2">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="135" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Some methods validate the arguments that are passed to them.</source>
          <target state="translated">Alguns métodos validam os argumentos que são passados a eles.</target>       </trans-unit>
        <trans-unit id="136" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>If they do and one of the arguments is <ph id="ph1">`null`</ph>, the method throws an <ph id="ph2">&lt;xref:System.ArgumentNullException?displayProperty=nameWithType&gt;</ph> exception.</source>
          <target state="translated">Se eles fazem e um dos argumentos for <ph id="ph1">`null`</ph>, o método gera uma <ph id="ph2">&lt;xref:System.ArgumentNullException?displayProperty=nameWithType&gt;</ph> exceção.</target>       </trans-unit>
        <trans-unit id="137" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Otherwise, it throws a <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exception.</source>
          <target state="translated">Caso contrário, ele gera um <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> exceção.</target>       </trans-unit>
        <trans-unit id="138" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following example illustrates this scenario.</source>
          <target state="translated">O exemplo a seguir ilustra esse cenário.</target>       </trans-unit>
        <trans-unit id="139" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>To address this issue, make sure that the argument passed to the method is not <ph id="ph1">`null`</ph>, or handle the thrown exception in a <ph id="ph2">`try…catch…finally`</ph> block.</source>
          <target state="translated">Para resolver esse problema, certifique-se de que o argumento passado para o método não é <ph id="ph1">`null`</ph>, ou lidar com a exceção lançada em um <ph id="ph2">`try…catch…finally`</ph> bloco.</target>       </trans-unit>
        <trans-unit id="140" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For more information, see <bpt id="p1">[</bpt>Exceptions<ept id="p1">](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735)</ept>.</source>
          <target state="translated">Para obter mais informações, consulte <bpt id="p1">[</bpt>exceções<ept id="p1">](http://msdn.microsoft.com/library/f99a1d29-a2a8-47af-9707-9909f9010735)</ept>.</target>       </trans-unit>
        <trans-unit id="141" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>The following Microsoft intermediate language (MSIL) instructions throw <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>: <ph id="ph2">`callvirt`</ph>, <ph id="ph3">`cpblk`</ph>, <ph id="ph4">`cpobj`</ph>, <ph id="ph5">`initblk`</ph>, <ph id="ph6">`ldelem.&lt;type&gt;`</ph>, <ph id="ph7">`ldelema`</ph>, <ph id="ph8">`ldfld`</ph>, <ph id="ph9">`ldflda`</ph>, <ph id="ph10">`ldind.&lt;type&gt;`</ph>, <ph id="ph11">`ldlen`</ph>, <ph id="ph12">`stelem.&lt;type&gt;`</ph>, <ph id="ph13">`stfld`</ph>, <ph id="ph14">`stind.&lt;type&gt;`</ph>, <ph id="ph15">`throw`</ph>, and <ph id="ph16">`unbox`</ph>.</source>
          <target state="translated">O Microsoft seguir geram instruções de linguagem intermediária (MSIL) <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>: <ph id="ph2">`callvirt`</ph>, <ph id="ph3">`cpblk`</ph>, <ph id="ph4">`cpobj`</ph>, <ph id="ph5">`initblk`</ph>, <ph id="ph6">`ldelem.&lt;type&gt;`</ph>, <ph id="ph7">`ldelema`</ph>, <ph id="ph8">`ldfld`</ph>, <ph id="ph9">`ldflda`</ph>, <ph id="ph10">`ldind.&lt;type&gt;`</ph> , <ph id="ph11">`ldlen`</ph>, <ph id="ph12">`stelem.&lt;type&gt;`</ph>, <ph id="ph13">`stfld`</ph>, <ph id="ph14">`stind.&lt;type&gt;`</ph>, <ph id="ph15">`throw`</ph>, e <ph id="ph16">`unbox`</ph>.</target>       </trans-unit>
        <trans-unit id="142" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source><ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> uses the HRESULT COR_E_NULLREFERENCE, which has the value 0x80004003.</source>
          <target state="translated"><ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph> usa o HRESULT COR_E_NULLREFERENCE, que tem o valor 0x80004003.</target>       </trans-unit>
        <trans-unit id="143" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For a list of initial property values for an instance of <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>, see the <ph id="ph2">&lt;xref:System.NullReferenceException.%23ctor%2A&gt;</ph> constructors.</source>
          <target state="translated">Para obter uma lista de valores de propriedade inicial para uma instância do <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>, consulte o <ph id="ph2">&lt;xref:System.NullReferenceException.%23ctor%2A&gt;</ph> construtores.</target>       </trans-unit>
        <trans-unit id="144" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source><bpt id="p1">**</bpt>Handling NullReferenceException in release code<ept id="p1">**</ept></source>
          <target state="translated"><bpt id="p1">**</bpt>Tratamento de NullReferenceException no código da versão<ept id="p1">**</ept></target>       </trans-unit>
        <trans-unit id="145" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>It's usually better to avoid a NullReferenceException than to handle it after it occurs.</source>
          <target state="translated">É melhor evitar uma NullReferenceException para tratá-la depois que ele ocorre.</target>       </trans-unit>
        <trans-unit id="146" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Handling an exception can make your code harder to maintain and understand, and can sometimes introduce other bugs.</source>
          <target state="translated">Tratar uma exceção pode fazer com que seu código seja difícil de manter e entender e pode, algumas vezes, introduzir outros bugs.</target>       </trans-unit>
        <trans-unit id="147" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>A NullReferenceException is often a non-recoverable error.</source>
          <target state="translated">Uma NullReferenceException é frequentemente um erro não recuperável.</target>       </trans-unit>
        <trans-unit id="148" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>In these cases, letting the exception stop the app might be the best alternative.</source>
          <target state="translated">Nesses casos, deixar a exceção interromper o aplicativo pode ser a melhor alternativa.</target>       </trans-unit>
        <trans-unit id="149" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>However, there are many situations where handling the error can be useful:</source>
          <target state="translated">No entanto, há várias situações em que o tratamento do erro pode ser útil:</target>       </trans-unit>
        <trans-unit id="150" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Your app can ignore objects that are null.</source>
          <target state="translated">Seu aplicativo pode ignorar objetos nulos.</target>       </trans-unit>
        <trans-unit id="151" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For example, if your app retrieves and processes records in a database, you might be able to ignore some number of bad records that result in null objects.</source>
          <target state="translated">Por exemplo, se seu aplicativo recupera e processa gravações em um banco de dados, você pode ser capaz de ignorar um certo número de gravações ruins que resultam em objetos nulos.</target>       </trans-unit>
        <trans-unit id="152" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Recording the bad data in a log file or in the application UI might be all you have to do.</source>
          <target state="translated">Gravar os dados ruins em um arquivo de log ou na interface de usuário do aplicativo pode ser tudo que você precisa fazer.</target>       </trans-unit>
        <trans-unit id="153" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You can recover from the exception.</source>
          <target state="translated">Você pode recuperar a partir da exceção.</target>       </trans-unit>
        <trans-unit id="154" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For example, a call to a web service that returns a reference type might return null if the connection is lost or the connection times out. You can attempt to reestablish the connection and try the call again.</source>
          <target state="translated">Por exemplo, uma chamada para um serviço Web que retorna um tipo de referência pode retornar nulo se a conexão for perdida ou expirar. Você pode tentar restabelecer a conexão e tentar a chamada de novo.</target>       </trans-unit>
        <trans-unit id="155" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You can restore the state of your app to a valid state.</source>
          <target state="translated">Você pode restaurar o estado de seu aplicativo para um estado válido.</target>       </trans-unit>
        <trans-unit id="156" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For example, you might be performing a multi-step task that requires you to save information to a data store before you call a method that throws a NullReferenceException.</source>
          <target state="translated">Por exemplo, pode ser que você esteja realizando uma tarefa de várias etapas que requer que as informações sejam salvas em um armazenamento de dados antes de chamar um método que lança uma NullReferenceException.</target>       </trans-unit>
        <trans-unit id="157" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>If the uninitialized object would corrupt the data record, you can remove the previous data before you close the app.</source>
          <target state="translated">Se o objeto não inicializado corromper a gravação de dados, você poderá remover os dados anteriores antes de fechar o aplicativo.</target>       </trans-unit>
        <trans-unit id="158" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You want to report the exception.</source>
          <target state="translated">Você deseja reportar a exceção.</target>       </trans-unit>
        <trans-unit id="159" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>For example, if the error was caused by a mistake from the user of your app, you can generate a message to help him supply the correct information.</source>
          <target state="translated">Por exemplo, se o erro foi causado por um erro do usuário de seu aplicativo, você pode gerar uma mensagem para ajudá-lo a fornecer as informações corretas.</target>       </trans-unit>
        <trans-unit id="160" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>You can also log information about the error to help you fix the problem.</source>
          <target state="translated">Você pode também registrar informações sobre o erro para ajudá-lo a solucionar o problema.</target>       </trans-unit>
        <trans-unit id="161" translate="yes" xml:space="preserve" extradata="MT" uid="T:System.NullReferenceException">
          <source>Some frameworks, like ASP.NET, have a high-level exception handler that captures all errors to that the app never crashes; in that case, logging the exception might be the only way you can know that it occurs.</source>
          <target state="translated">Algumas estruturas, como o ASP.NET, tem um manipulador de exceção de alto nível que captura todos os erros para que o aplicativo nunca falhe; nesse caso, a exceção de registro em log pode ser a única maneira de saber que ele ocorre.</target>       </trans-unit>
        <trans-unit id="162" translate="yes" xml:space="preserve" uid="T:System.NullReferenceException">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class.</source>
          <target state="translated">Inicializa uma nova instância da classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="163" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class, setting the <ph id="ph2">&lt;see cref="P:System.Exception.Message" /&gt;</ph> property of the new instance to a system-supplied message that describes the error, such as "The value 'null' was found where an instance of an object was required."</source>
          <target state="translated">Inicializa uma nova instância da classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph>, definindo a propriedade <ph id="ph2">&lt;see cref="P:System.Exception.Message" /&gt;</ph> da nova instância para uma mensagem fornecida pelo sistema que descreve o erro, como "o valor 'null' foi encontrado quando uma instância de um objeto foi solicitada."</target>       </trans-unit>
        <trans-unit id="164" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor">
          <source>This message takes into account the current system culture.</source>
          <target state="translated">Esta mensagem leva em conta a cultura do sistema atual.</target>       </trans-unit>
        <trans-unit id="165" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source>The following table shows the initial property values for an instance of <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</source>
          <target state="translated">A tabela a seguir mostra os valores de propriedade iniciais para uma instância de <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="166" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source>Property</source>
          <target state="translated">Propriedade</target>       </trans-unit>
        <trans-unit id="167" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source>Value</source>
          <target state="translated">Valor</target>       </trans-unit>
        <trans-unit id="168" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source><ph id="ph1">`null`</ph>.</source>
          <target state="translated"><ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="169" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor">
          <source>The localized error message string.</source>
          <target state="translated">A cadeia de caracteres da mensagem de erro localizada.</target>       </trans-unit>
        <trans-unit id="170" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>A <ph id="ph1">&lt;see cref="T:System.String" /&gt;</ph> that describes the error.</source>
          <target state="translated">Um <ph id="ph1">&lt;see cref="T:System.String" /&gt;</ph> que descreve o erro.</target>       </trans-unit>
        <trans-unit id="171" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>The content of <bpt id="p1">&lt;c&gt;</bpt>message<ept id="p1">&lt;/c&gt;</ept> is intended to be understood by humans.</source>
          <target state="translated">O conteúdo da <bpt id="p1">&lt;c&gt;</bpt>mensagem<ept id="p1">&lt;/c&gt;</ept> deve ser compreendido por humanos.</target>       </trans-unit>
        <trans-unit id="172" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>The caller of this constructor is required to ensure that this string has been localized for the current system culture.</source>
          <target state="translated">O chamador desse construtor é necessário para garantir que essa cadeia de caracteres tenha sido localizada para a cultura do sistema atual.</target>       </trans-unit>
        <trans-unit id="173" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class with a specified error message.</source>
          <target state="translated">Inicializa uma nova instância da classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> com uma mensagem de erro especificada.</target>       </trans-unit>
        <trans-unit id="174" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>The following table shows the initial property values for an instance of <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</source>
          <target state="translated">A tabela a seguir mostra os valores de propriedade iniciais para uma instância de <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="175" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>Property</source>
          <target state="translated">Propriedade</target>       </trans-unit>
        <trans-unit id="176" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>Value</source>
          <target state="translated">Valor</target>       </trans-unit>
        <trans-unit id="177" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source><ph id="ph1">`null`</ph>.</source>
          <target state="translated"><ph id="ph1">`null`</ph>.</target>       </trans-unit>
        <trans-unit id="178" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String)">
          <source>The error message string.</source>
          <target state="translated">A cadeia de caracteres da mensagem de erro.</target>       </trans-unit>
        <trans-unit id="179" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>The object that holds the serialized object data.</source>
          <target state="translated">O objeto que mantém os dados de objeto serializados.</target>       </trans-unit>
        <trans-unit id="180" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>The contextual information about the source or destination.</source>
          <target state="translated">As informações contextuais sobre a origem ou o destino.</target>       </trans-unit>
        <trans-unit id="181" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class with serialized data.</source>
          <target state="translated">Inicializa uma nova instância da classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> com dados serializados.</target>       </trans-unit>
        <trans-unit id="182" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</source>
          <target state="translated">Este construtor é chamado durante a desserialização para reconstituir o objeto de exceção transmitido por um fluxo.</target>       </trans-unit>
        <trans-unit id="183" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)">
          <source>For more information, see <bpt id="p1">[</bpt>XML and SOAP Serialization<ept id="p1">](~/docs/standard/serialization/xml-and-soap-serialization.md)</ept>.</source>
          <target state="translated">Para obter mais informações, consulte <bpt id="p1">[</bpt>XML e serialização SOAP<ept id="p1">](~/docs/standard/serialization/xml-and-soap-serialization.md)</ept>.</target>       </trans-unit>
        <trans-unit id="184" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The error message that explains the reason for the exception.</source>
          <target state="translated">A mensagem de erro que explica a razão da exceção.</target>       </trans-unit>
        <trans-unit id="185" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The exception that is the cause of the current exception.</source>
          <target state="translated">A exceção que é a causa da exceção atual.</target>       </trans-unit>
        <trans-unit id="186" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>If the <bpt id="p1">&lt;c&gt;</bpt>innerException<ept id="p1">&lt;/c&gt;</ept> parameter is not <ph id="ph1">&lt;see langword="null" /&gt;</ph>, the current exception is raised in a <ph id="ph2">&lt;see langword="catch" /&gt;</ph> block that handles the inner exception.</source>
          <target state="translated">Se o parâmetro <bpt id="p1">&lt;c&gt;</bpt>innerException<ept id="p1">&lt;/c&gt;</ept> não for <ph id="ph1">&lt;see langword="null" /&gt;</ph>, a exceção atual será gerada em um bloco <ph id="ph2">&lt;see langword="catch" /&gt;</ph> que tratará a exceção interna.</target>       </trans-unit>
        <trans-unit id="187" translate="yes" xml:space="preserve" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>Initializes a new instance of the <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> class with a specified error message and a reference to the inner exception that is the cause of this exception.</source>
          <target state="translated">Inicializa uma nova instância da classe <ph id="ph1">&lt;see cref="T:System.NullReferenceException" /&gt;</ph> com uma mensagem de erro especificada e uma referência à exceção interna que é a causa da exceção.</target>       </trans-unit>
        <trans-unit id="188" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <ph id="ph1">&lt;xref:System.Exception.InnerException%2A&gt;</ph> property.</source>
          <target state="translated">Uma exceção que é lançada como um resultado direto de uma exceção anterior deve incluir uma referência para a exceção anterior na propriedade <ph id="ph1">&lt;xref:System.Exception.InnerException%2A&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="189" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The <ph id="ph1">&lt;xref:System.Exception.InnerException%2A&gt;</ph> property returns the same value that is passed into the constructor, or <ph id="ph2">`null`</ph> if the <ph id="ph3">&lt;xref:System.Exception.InnerException%2A&gt;</ph> property does not supply the inner exception value to the constructor.</source>
          <target state="translated">A propriedade <ph id="ph1">&lt;xref:System.Exception.InnerException%2A&gt;</ph> retorna o mesmo valor que é passado para o construtor, ou <ph id="ph2">`null`</ph> caso a propriedade <ph id="ph3">&lt;xref:System.Exception.InnerException%2A&gt;</ph> não forneça o valor de exceção interno para o construtor.</target>       </trans-unit>
        <trans-unit id="190" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The following table shows the initial property values for an instance of <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</source>
          <target state="translated">A tabela a seguir mostra os valores de propriedade iniciais para uma instância de <ph id="ph1">&lt;xref:System.NullReferenceException&gt;</ph>.</target>       </trans-unit>
        <trans-unit id="191" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>Property</source>
          <target state="translated">Propriedade</target>       </trans-unit>
        <trans-unit id="192" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>Value</source>
          <target state="translated">Valor</target>       </trans-unit>
        <trans-unit id="193" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The inner exception reference.</source>
          <target state="translated">A referência de exceção interna.</target>       </trans-unit>
        <trans-unit id="194" translate="yes" xml:space="preserve" extradata="MT" uid="M:System.NullReferenceException.#ctor(System.String,System.Exception)">
          <source>The error message string.</source>
          <target state="translated">A cadeia de caracteres da mensagem de erro.</target>       </trans-unit>
      </group>
    </body>
  </file>
</xliff>